apps/sports-helsinki/src/domain/search/eventSearch/SearchPage.tsx

```diff
@@ -18,27 +20,34 @@ import { toast } from 'react-toastify';
 import EventList from '../../../common-events/components/eventList/EventList';
 import { SEARCH_ROUTES } from '../../../constants';
 import { removeQueryParamsFromRouter } from '../../../utils/routerUtils';
+import type {
+  CombinedSearchProps,
+  SearchResultItemsCount,
+} from '../combinedSearch/CombinedSearch';
 import { EVENT_SORT_OPTIONS, PAGE_SIZE } from './constants';
 import styles from './eventSearchPage.module.scss';
 import SearchResultsContainer from './searchResultList/SearchResultsContainer';
 import { getEventSearchVariables, getNextPage } from './utils';
 
 const SearchPage: React.FC<{
-  SearchComponent: React.FC<{
-    scrollToResultList: () => void;
-    'data-testid'?: string;
-  }>;
+  SearchComponent: React.FC<CombinedSearchProps>;
   pageTitle: string;
 }> = ({ SearchComponent, pageTitle }) => {
   const { t } = useSearchTranslation();
   const router = useRouter();
-  const params: { place?: string } = router.query;
-
   const [isFetchingMore, setIsFetchingMore] = React.useState(false);
   const isSmallScreen = useIsSmallScreen();
-
   const { meta } = useConfig();
 
+  const params: { place?: string; eventType?: string } = router.query;
+  const eventTypes = Object.keys(EventTypeId);
+  const primarySearchEventType = eventTypes.find(
+    (e) => (params.eventType as EventTypeId) === e
+  ) as EventTypeId;
+  const secondarySearchEventType = eventTypes.find(
```

Comment:
looks like it works only when there are only 2 types in array, which is our case:)

Reply:
True :) I thought about that too, but didn't want to overcomplicate it.